lib/pbench/agent/tool_meister.py

```diff
@@ -1518,6 +1533,39 @@ def end_tools(self, data):
                     name,
                 )
                 failures += 1
+
+        # Remove persistent tool temporary working directory on remote host
+        directory = data["directory"]
+        tool_dir = self.directories[directory]
+
+        if self._hostname == self._controller:
+            directory_to_delete = tool_dir
+        else:
+            directory_to_delete = tool_dir.parent
```

Comment:
This seems like a bad idea:  doing it this way spreads specialize knowledge across multiple areas of the code when it could (and should) be encapsulated in one.  We should be recording the directory that we want deleted in `init_tools()` so that we _don't_ have to be making adjustments like this, here.

Reply:
Yep I concur that we should be recording the directory that we want deleted in init_tools() itself.